src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -544,6 +544,13 @@ internal virtual void ValidateParameters()
                 ErrorRecord error = GetValidationError(WebCmdletStrings.OutFileMissing, "WebCmdletOutFileMissingException", nameof(Resume));
                 ThrowTerminatingError(error);
             }
+
+            // OutFile must not be a directory to use Resume.
+            if (Resume.IsPresent && Directory.Exists(QualifiedOutFile))
```

Comment:
1. If we resolve the path we should cache it here and use later.
2. It seems it is not correct check since we could get the file name from URL.

Reply:
1. I agree with you, I'll do it later
2. I think it's correct, this is the logic:
- if -Outfile points to a file, old logic you can use resume
- if -Outfile points to a folder (Directory.Exists(QualifiedOutFile)) you can't use the resume switch and the filename is taken later from the url

Reply:
@iSazonov I think I need a hand with caching the resolved path

Reply:
> * if -Outfile points to a folder (Directory.Exists(QualifiedOutFile)) you can't use the resume switch (throws an error) and the filename is taken later from the url

If we take a file name from URL we can use Resume, can't we?